src/Dapplo.Microsoft.Extensions.Hosting.AppServices/ResourceNamedPipeServer.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Dapplo and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO.Pipes;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+#if NET472
+using System.Security.AccessControl;
+using System.Security.Principal;
+#endif
+
+namespace Dapplo.Microsoft.Extensions.Hosting.AppServices
+{
+    /// <summary>
+    ///     This class is the server resource for a named pipe
+    /// </summary>
+    public sealed class ResourceNamedPipeServer : BaseResourceNamedPipe, IDisposable
+    {
+        private byte[] _buffer = new byte[32];
+        private object _lockingObject = new object();
+        private bool _isStopping = false;
+        private NamedPipeServerStream _applicationNamedPipe;
+
+        private ResourceNamedPipeServer(ILogger logger, string namedPipeId, string resourceName = null) : base(logger, namedPipeId, resourceName)
+        {
+        }
+
+        /// <summary>
+        ///   This event is fired when a named pipe client is connected
+        /// </summary>
+        public event EventHandler Connected;
+
+        /// <summary>
+        ///     Create a ResourceNamedPipeServer for the specified named pipe id and resource-name
+        /// </summary>
+        /// <param name="logger">ILogger</param>
+        /// <param name="namedPipeId">ID of the named pipe, preferably a Guid as string</param>
+        /// <param name="resourceName">Name of the resource to lock, e.g your application name, useful for logs</param>
+        /// <param name="global">true to have a global named pipe see: https://msdn.microsoft.com/en-us/library/bwe34f1k.aspx</param>
+        public static ResourceNamedPipeServer Create(ILogger logger, string namedPipeId, string resourceName = null, bool global = false)
```

Comment:
I don't think I like to enforce the complexity of the Mutex, with global/local in this class. It should know as little as possible.